### ISSUE TITLE
[charts] PieChart disappears with certain data values

### ISSUE THREAD

**Issue by samwato** [mui/mui-x#14167]
### Steps to reproduce

Link to live example: https://codesandbox.io/p/sandbox/epic-bush-4jk7jf

This issue may have been raised before, however, the example shows a pie chart hidden and 2 pie charts that are visible.

![Screenshot from 2024-08-12 10-10-51](https://github.com/user-attachments/assets/1ab8f053-9362-4fe2-81f8-da1ca28fdfcc)

### Current behavior

The pie chart is hidden

### Expected behavior

The pie chart is visible

### Context

My use case is the data values are dynamic and have ran into this issue by accident

**Search keywords**: piechart

**Comment by alexfauquette** [comment]
It's the first time we see this issue.

Seems to come from the computation of the arc, because the path value started with `M0,-77.5A` for the invisible arc. But starts with `M0.001,-77.5A` for others.

I still need to investigate if this rounding issue come from the animation interpolation not finishing on the correct value, or if it's the d3.arc() that has an issue with this extreme edge case

**Comment by JCQuintas** [comment]
Seems to have been fixed along the way https://stackblitz.com/edit/qsuqxk5t?file=src%2FDemo.tsx

**Comment by alexfauquette** [comment]
@JCQuintas That would be weird. The issue just moved because we modified the computation of the drawing area. Now the limit point is when the chart size moves from 170 to 169

https://stackblitz.com/edit/qsuqxk5t-kcluagbs?file=src%2FDemo.tsx

**Comment by mustafajw07** [comment]
Hi @alexfauquette , I looked into this and it seems the issue comes from arc generation (likely d3) when values are extremely small, producing a zero-length path that doesn’t render.

A possible fix could be to clamp very small values with a minimum threshold or ignore values below a limit to avoid degenerate arcs.

If this approach makes sense, I’d be happy to work on a fix/PR. Let me know your thoughts.

**Comment by alexfauquette** [comment]
The solution looks good. It's fairly similar to the one proposed  in https://github.com/mui/mui-x/issues/14167#issuecomment-2284108778

The remaining issue is: How to let users define which behavior will apply on their chart

**Comment by mustafajw07** [comment]
> The solution looks good. It's fairly similar to the one proposed in [#14167 (comment)](https://github.com/mui/mui-x/issues/14167#issuecomment-2284108778)
> 
> The remaining issue is: How to let users define which behavior will apply on their chart

Good point. We could expose a prop to control how small values are handled (e.g. ignore vs clamp to a minimum angle).
That would let users decide between visual accuracy and visibility depending on their needs.
